File: externals/Async-CTP/samples/VB/Windows/Phone/Concert/Finder/Concert/Finder/Model/ISearchable.vb
Imports System
Imports System.ComponentModel
Imports System.Xml.Serialization

Namespace ConcertFinder.Model
    ''' <summary>
    ''' Represents a search result.
    ''' </summary>
    Public Interface ISearchable

        ''' <summary>
        ''' The ID of the search result.
        ''' </summary>
        Property Id As String

        ''' <summary>
        ''' The name of the search result.
        ''' </summary>
        Property Name As String

        ''' <summary>
        ''' The image for the search result.
        ''' </summary>
        <XmlIgnore()>
        Property Image As Uri

        <EditorBrowsable(EditorBrowsableState.Never)>
        Property _Image As String

        ''' <summary>
        ''' The Uri of the search result.
        ''' </summary>
        <XmlIgnore()>
        Property Uri As Uri

        <EditorBrowsable(EditorBrowsableState.Never)>
        Property _Uri As String

    End Interface

End Namespace

File: externals/Async-CTP/samples/CS/Windows/Phone/Concert/Finder/Concert/Finder/Model/ISearchable.cs
using System;
using System.ComponentModel;
using System.Xml.Serialization;

namespace ConcertFinder.Model
{
    /// <summary>
    /// Represents a search result.
    /// </summary>
    public interface ISearchable
    {
        /// <summary>
        /// The ID of the search result.
        /// </summary>
        string Id { get; set; }

        /// <summary>
        /// The name of the search result.
        /// </summary>
        string Name { get; set; }

        /// <summary>
        /// The image for the search result.
        /// </summary>
        [XmlIgnore]
        Uri Image { get; set; }
        [EditorBrowsable(EditorBrowsableState.Never)]
        string _Image { get; set; }

        /// <summary>
        /// The Uri of the search result.
        /// </summary>
        [XmlIgnore]
        Uri Uri { get; set; }
        [EditorBrowsable(EditorBrowsableState.Never)]
        string _Uri { get; set; }
    }
}
